drivers/src/nrfx_saadc.c

```diff
@@ -626,4 +626,21 @@ void nrfx_saadc_limits_set(uint8_t channel, int16_t limit_low, int16_t limit_hig
         nrf_saadc_int_enable(int_mask);
     }
 }
+
+void nrfx_enable_adc_chan(int chan, nrf_saadc_input_t   pselp,
+                          nrf_saadc_input_t             pseln)
+{
+    NRFX_ASSERT(m_cb.active_channels < NRF_SAADC_CHANNEL_COUNT);
```

Comment:
I think it could be good to check with assert if nrfx_saadc driver is still initialized:
`NRFX_ASSERT(m_cb.state != NRFX_DRV_STATE_UNINITIALIZED);`